### PR TITLE
Update release workflow

### PR DESCRIPTION
**Pull request by finn-tbd**
This fixes a number of issues with the previous release workflow.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -229,6 +231,7 @@ jobs:
           path: bound/kt/src/main/resources/
 
       - name: Deploy Release Version of Kotlin Project to Maven Central
+        if: ${{ env.SONATYPE_PASSWORD != '' }}
```
Comment (nitro-neal) [inline .github/workflows/release.yaml:234]: we still need this?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -250,6 +253,69 @@ jobs:
           SIGN_KEY_PASS: ${{ secrets.GPG_SECRET_PASSPHRASE }}
           SIGN_KEY: ${{ secrets.GPG_SECRET_KEY }}
 
+      - name: Create GitHub Release
+        uses: actions/create-release@0cb9c9b65d5d1901c1f53e5e66eaf4afd303e70e # v1.1.4 final release
```
Comment (nitro-neal) [inline .github/workflows/release.yaml:257]:
```suggestion
        uses: actions/create-release@0cb9c9b65d5d1901c1f53e5e66eaf4afd303e70e
```

**Review by nitro-neal — APPROVED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (finn-tbd) [reply .github/workflows/release.yaml:234]: not really but it will be useful if we want to debug this in the future.

Comment (finn-tbd) [reply .github/workflows/release.yaml]: Had to update some other stuff so I took this out too.

**Pull request merged**